client.go

```diff
@@ -658,6 +660,31 @@ func (c *Client) SetTLSClientConfig(config *tls.Config) *Client {
 	return c
 }
 
+// SetResolver method set Custom DNS for resolver
```

Comment:
The comment should be more informative IMO. please include an example and also address the fact that this setting will be overridden if SetTransport is called after, like so:
`c.SetResolver("foo").SetTransport(someOtherTransport)`